src/main/perf/TaskParser.java

```diff
@@ -81,52 +81,139 @@ public TaskParser(IndexState state,
+      List<String> drillDowns = parseDrillDowns();
+      doStoredLoadsTask = TaskParser.this.doStoredLoads;
+      parseHilite();
+      String[] taskAndType = parseTaskType(text);
+      String taskType = taskAndType[0];
+      text = taskAndType[1];
+      Query query = buildQuery(taskType, text);
+      Query query2 = applyDrillDowns(query, drillDowns);
+      Query query3 = applyFilter(query2, filter);
+      return new SearchTask(category, query2, sort, group, topN, doHilite, doStoredLoads, facets, doDrillSideways);
     }
-    final String category = line.substring(0, spot);
 
-    int spot2 = line.indexOf(" #");
-    if (spot2 == -1) {
-      spot2 = line.length();
+    String[] parseTaskType(String line) {
+      int spot = line.indexOf("//");
+      if (spot == -1) {
+        return new String[]{"", line};
+      } else {
+        return new String[]{
+          line.substring(0, spot),
+          line.substring(spot + 2)
+        };
+      }
     }
 
-    String text = line.substring(spot+1, spot2).trim();
-    String origText = text;
+    Query applyFilter(Query query, Query filter) {
+      if (filter == null) {
+        return query;
+      } else {
+        return new BooleanQuery.Builder()
+          .add(query, Occur.MUST)
+          .add(filter, Occur.FILTER)
+          .build();
+      }
+    }
+
+    Query applyDrillDowns(Query query, List<String> drillDowns) {
+      if (drillDowns.isEmpty()) {
+        return query;
+      }
+      DrillDownQuery q = new DrillDownQuery(state.facetsConfig, query);
+      for(String s : drillDowns) {
+        int i = s.indexOf('=');
+        if (i == -1) {
+          throw new IllegalArgumentException("drilldown is missing =");
+        }
+        String dim = s.substring(0, i);
+        String values = s.substring(i+1);
 
-    final Task task;
-    if (category.equals("Respell")) {
-      task = new RespellTask(new Term(fieldName, text));
-    } else {
-      if (text.length() == 0) {
-        throw new RuntimeException("null query line");
+        while (true) {
+          i = values.indexOf(',');
+          if (i == -1) {
+            q.add(dim, values);
+            break;
+          }
+          q.add(dim, values.substring(0, i));
+          values = values.substring(i+1);
+        }
       }
+      return q;
+    }
 
+    Query parseFilter() {
       // Check for filter (eg: " +filter=0.5%")
       final Matcher m = filterPattern.matcher(text);
-      Query filter;
       if (m.find()) {
         final double filterPct = Double.parseDouble(m.group(1));
         // Splice out the filter string:
         text = (text.substring(0, m.start(0)) + text.substring(m.end(0), text.length())).trim();
-        filter = new RandomQuery(filterPct);
-      } else {
-        filter = null;
+        return new RandomQuery(filterPct);
       }
+      return null;
+    }
 
+    void parseMinShouldMatch() {
       final Matcher m2 = minShouldMatchPattern.matcher(text);
       final int minShouldMatch;
```

Comment:
oops I think this shadows the class-level declaration. That kind of thing's why this is all a little scary.